go-controller/pkg/ovn/pods.go

```diff
@@ -109,6 +111,31 @@ func (oc *Controller) syncPodsRetriable(pods []interface{}) error {
 		if err != nil {
 			return fmt.Errorf("could not generate ops to delete stale ports from logical switch %s (%+v)", n.Name, err)
 		}
+
+		// have to set up the hybrid Overlay interface address here because all the pod IPs have been allocated above
+		if config.HybridOverlay.Enabled {
+			//check the annotation
+			nodeHybridOverlayIFAddrs := n.Annotations[hotypes.HybridOverlayIFAddress]
+			var hybridOverlayIFAddrs []string
+			if len(nodeHybridOverlayIFAddrs) > 0 {
+				hybridOverlayIFAddrs = strings.Split(nodeHybridOverlayIFAddrs, ",")
+
+			}
+			AllocatedHybridOverlayIFAddresses, err := oc.lsManager.AllocateHybridOverlay(n.Name, hybridOverlayIFAddrs)
+			if err != nil {
+				return fmt.Errorf("cannot Allocate hybrid overlay interface addresses: %v", err)
+			}
+			var sliceHybridOverlayIFAddrs []string
+			for _, ip := range AllocatedHybridOverlayIFAddresses {
+				sliceHybridOverlayIFAddrs = append(sliceHybridOverlayIFAddrs, ip.IP.String())
+			}
+			err = oc.kube.SetAnnotationsOnNode(n.Name, map[string]interface{}{hotypes.HybridOverlayIFAddress: strings.Join(sliceHybridOverlayIFAddrs, ",")})
```

Comment:
how could we ever have more than 1 address here?

Reply:
so I don't think that as of now it will ever be more then 1 ip address but I did it for a few reasons. Eventually we will want to support dual stack hybrid overlay and although it isn't supported right now it is possible that a node will have more then one hostSubnet and `AllocateIPs()` returns an address from all nodes subnets.

Reply:
We'd only ever care about dual-stack and thus have 2 max.

Never gonna have more than that, because it's really just a fake internal address (like our 100.64 router port addresses) that we use internally in OVN for L3 routing.